delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/model/CompositeView.scala

```diff
@@ -74,17 +76,55 @@ final case class CompositeView(
     * Looks for a source with the given identifier
     */
   def sources(sourceId: Iri): Either[SourceNotFound, CompositeViewSource] =
-    sources.value.find(_.id == sourceId).toRight(SourceNotFound(id, sourceId, project))
+    sources.lookup(sourceId).toRight(SourceNotFound(id, sourceId, project))
 
   /**
     * Looks for a projection with the given identifier
     */
   def projections(projectionId: Iri): Either[ProjectionNotFound, CompositeViewProjection] =
-    projections.value.find(_.id == projectionId).toRight(ProjectionNotFound(id, projectionId, project))
+    projections.lookup(projectionId).toRight(ProjectionNotFound(id, projectionId, project))
+
+  /**
+    * Looks for Elasticsearch projections
+    */
+  def elasticSearchProjections: Set[ElasticSearchProjection] = projectionsOf[ElasticSearchProjection]
+
+  /**
+    * Looks for Sparql projections
+    */
+  def sparqlProjections: Set[SparqlProjection] = projectionsOf[SparqlProjection]
+
+  private def projectionsOf[X: ClassTag] =
```

Comment:
It won't looks super nice but it is possible to avoid reflection for `sparqlProjections` and `elasticSearchProjections`. Probably doesn't really matter though